### DISCUSSION TITLE
How to change OS audio volume

### DISCUSSION THREAD

**Discussion by the-lightstack** [unikraft/unikraft#402] [Q&A] [answered]
I am not sure, which audio manager the Unikernel is using (if any, I hope so) therefore my question is: how can I change the operating system's master audio volume (would amixer work or do I have to do it on the pulseaudio level?)
(btw, I am playing the sounds with rust's rodio crate)
Thanks for your help

**Comment by nderjung** [comment, marked as answer]
Hi @the-lightstack,

Unikraft does not support playing audio, it does not have any audio drivers nor any way to control audio.  Unikraft facilitates the runtime of (commonly) server-based applications, like NGINX, as a purpose-built VM image.  If you are requiring to play audio, you will need to attach that as a device to the VM and then also have the relevant drivers for it.  For your usecase, you'd have to investigate this driver integration as an external library.  [Here is how it's done in the Linux kernel, for example](https://git.kernel.org/pub/scm/linux/kernel/git/tiwai/sound.git/commit/?h=for-next&id=de3a9980d8c34b2479173e809afa820473db676a).

For now, there is no plan to support pulseaudio or introduce audio drivers into Unikraft or its ecosystem at the time of writing by the core team.

**Comment by razvand** [comment]
@the-lightstack, since we are already (viciously) using [the Unikraft Discord server](https://bit.ly/UnikraftDiscord) for discussions, meetings, brainstorming and jokes (mostly bad, we do what we can), we will close the Discussions tab. We will do this on Friday, March 31, 2023.

Please move any discussions or questions on [the Unikraft Discord server](https://bit.ly/UnikraftDiscord). See you all there! :handshake:
